File: src/main/java/aginsun/kingdoms/server/entities/EntityBankerKeeper.java
package aginsun.kingdoms.server.entities;

import aginsun.kingdoms.api.entities.EntityNPC;
import aginsun.kingdoms.server.TaleOfKingdoms;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.util.EnumHand;
import net.minecraft.util.text.TextComponentString;
import net.minecraft.world.World;

import static aginsun.kingdoms.server.handlers.GuiHandler.GUI_BANK;

public final class EntityBankerKeeper extends EntityNPC
{
    public EntityBankerKeeper(World world)
    {
        super(world, new ItemStack(Items.BOOK), 40.0F);
    }

    @Override
    public boolean canBePushed()
    {
        return false;
    }

    @Override
    protected boolean isMovementBlocked()
    {
        return true;
    }

    @Override
    protected boolean processInteract(EntityPlayer player, EnumHand hand)
    {
        if (this.canInteractWith(player))
        {
            if (!this.world.isRemote)
            {
                this.heal(100.0F);
                player.sendMessage(new TextComponentString("npc.banker.dialog.welcome"));
            }
            player.openGui(TaleOfKingdoms.instance, GUI_BANK, world, 0, 0, 0);
        }
        return true;
    }
}

File: src/main/java/kingdoms/server/entities/EntityBankerKeeper.kt
package kingdoms.server.entities

import kingdoms.api.entities.EntityNPC
import kingdoms.server.handlers.NetworkHandler
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.init.Items
import net.minecraft.item.ItemStack
import net.minecraft.util.ChatComponentTranslation
import net.minecraft.world.World

class EntityBankerKeeper(world: World): EntityNPC(world, ItemStack(Items.book), 40.0F)
{
    override fun canBePushed(): Boolean
    {
        return false
    }

    override fun isMovementCeased(): Boolean
    {
        return true
    }

    override fun interact(player: EntityPlayer): Boolean
    {
        if (!worldObj.isRemote)
        {
            if (this.canInteractWith(player))
            {
                this.heal(100.0F)
                player.addChatMessage(ChatComponentTranslation("npc.banker.dialog.welcome"))
                NetworkHandler.openGui(this.entityId, player)
            }
        }
        return false
    }
}